### ISSUE TITLE
Daily Template not detecting code in configuration file

### ISSUE THREAD

**Issue by abcitygit** [NotePlan/plugins#233]
**Describe the bug**
The Daily Template plugin isn't detecting it's code in the configuration file and keep asking to install a new code block.

**To Reproduce**
Here is the plug _configuration.md file. I removed the API keys.

#  _configuration
---
``` javascript
{
  dailyJournal: {
    templateTitle: 'Daily Note Template',
    reviewSectionHeading: 'Journal',
    moods: '🤩 Great,🙂 Good,😇 Blessed,🥱 Tired,😫 Stressed,😤 Frustrated,😔 Low,🥵 Sick,Other',
    reviewQuestions: '@sleep(<number>)\n@work(<number>)\n@fruitveg(<int>)\nMood:: <mood>\nExercise:: <string>\nGratitude:: <string>\nGod was:: <string>\nAlive:: <string>\nNot Great:: <string>\nWife:: <string>\nRemember:: <string>'
  },

}
```

``` javascript
{
		// configuration for daily quote (used in Daily Note Template, for example)
  quote: {
    // Available modes: [random (default), today, author]
    mode: 'today',
    // API key required, available authors: https://premium.zenquotes.io/available-authors/
    author: 'anne-frank',
    // Required for mode: 'author' (from https://premium.zenquotes.io/)
    apiKey: 'b1a83392827ee9da554ac1ccb1317a60b055cc7d',
  },
}
```

``` javascript
{
// configuration for weather data (used in Daily Note Template, for example)
  weather: {
    // API key for https://openweathermap.org/
    openWeatherAPIKey: 'Api', // !!REQUIRED!!
    // Required location for weather forecast
    latPosition: 00.00,  // !!REQUIRED!!
    longPosition: 00.00, // !!REQUIRED!!
    // Default units. Can be 'metric' (for Celsius), or 'imperial' (for Fahrenheit)
    openWeatherUnits: 'metric',
  },

}
```
# Template Tag Configuration
This note provides a central location where you can configure various plugin options:

- Use the fenced code block below (which comes after a line made by 3 dashes) to customize global values for the various template tags.

- NotePlan plugin configuration uses JSON5 [JSON5 | JSON for Humans](https://json5.org/), which is a human-friendly superset of JSON, providing things such as comments, unquoted keys and other common patterns available in standard JavaScript objects.

	*Note: Despite JSON5 enhancements, multi-line strings are not supported, therefore to include them you need to use "\n" (new line) for line breaks rather than actual line breaks.*

- Use the code block marked as `javascript` shown below to write your own custom custom configurations.

- While it is possible to have multiple `javascript` code blocks in this document, only the **first** code block will be used.

- If you have a quoted "string" which you want to have a **line break**, insert a "\n" (i.e. backslash-n) where you want the line break, e.g., "These\nAre\nThree lines".

### Validating Configuration
The configuration code blocks below are validated in **realtime** by NotePlan as you edit:

- If there is a configuration mistake, all code below will all be a single color (based on theme).
- If the configuration passes the validation, you will see configuration settings will be formatted based on your current theme (e.g., orange, green, purple, black, etc.).

**TIP:** If your configuration is invalid, you can copy/paste the configuration block to [JSON5 Validator Online - JSON5 lint Tool to validate JSON5 data](https://codebeautify.org/json5-validator) which will provide details about the error(s) in your code block, indicating which line(s) contain the error(s).

### Reporting Plugin Issues
Should you run into an issue with a NotePlan plugin, you can use one of the following methods (in order of priority)
- 🐞 [NotePlan Plugin Issues](https://github.com/NotePlan/plugins/issues/new/choose)
- 🧩 [Discord](https://discord.com/channels/763107030223290449/784376250771832843)
- 📪 [NotePlan Support](hello@noteplan.io)
*****
## Plugin Configuration
*Note: While the following code block is marked as `javascript` it is actually `JSON5` and is only marked as `javascript` so that your theme will provide the appropriate syntax highlighting.*

```javascript
{
  // Note Even though it is fenced as "javascript", this configuration is actually JSON5.

  // configuration for dates, heavily based on javascript's Intl module
  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat
{
  date: {
    // Default timezone for date and time.
    timezone: 'automatic',
    // Default locale to format date and time.
    // e.g. en-US will result in mm/dd/yyyy, while en_GB will be dd/mm/yyyy
    locale: 'en-US',
    // can be "short", "medium", "long" or "full"
    dateStyle: 'short',
    // optional key, can be "short", "medium", "long" or "full"
    timeStyle: 'short',
    // time format when invoking dwertheimer.DateAutomations/formatted
    // see https://www.strfti.me/ to aid in creating custom formats)
    format: '%Y-%m-%d %I:%M:%S %P'
  },}





{
  // default values for custom tags.
  // These tags cannot be functions, but you may choose to have nested objects.
  // feel free to edit this value however you see fit.
  tagValue: {
    me: {
      // Can be used as {{me.firstName}}
      firstName: 'John',
      // Can be used as {{me.lastName}}
      lastName: 'Doe',
    }
    // ...
  },}
}

```

**Expected behavior**
I keep trying to run the plugins but it never seems to detect the plugins that are in the configuration file even. I've removed the API keys from this.

**Screenshots**

**System**
 - Device: Mac
 - OS: macOS 12.1
 - NotePlan Version 3.3 (697)
 - Plugin: Daily Journal v0.11.0

**Additional context**

**Plugin Console Log**
JSLog: 
dayStart:
JSLog: Executing 'todayStart' took: 0.000762959 seconds
JSLog: Opened: 2022-02-11
JSLog:     validateMinimumConfig: passed minimum validation spec
JSLog: applyNamedTemplate: for template 'Daily Note Template'
JSLog: processTemplate() found tag quote()
JSLog: processTag(quote()) param:
JSLog: execTagListFunction() Tag matched "quote"
JSLog: getDailyQuote() with :
JSLog: 	Info: No 'quote' settings in Templates/_configuration note
JSLog: Before API call: https://zenquotes.io/api/random
JSLog: 	Don't take yourself too seriously, pretty soon you can find the humor in our everyday lives. - *Betty White*
JSLog: ---- execTagListFunction(quote) RESULT="Don't take yourself too seriously, pretty soon you can find the humor in our everyday lives. - *Betty White*"

JSLog: processTemplate() found tag weather()
JSLog: processTag(weather()) param:
JSLog: execTagListFunction() Tag matched "weather"
JSLog:   getOrMakeConfigurationSection: no 'weather' config section found
JSLog:   getOrMakeConfigurationSection: default available

**Comment by jgclark** [comment]
You're not running the latest version of NotePlan or the plugin. And the configuration blocks are each separate code blocks not one longer code block with three sections in it. See the example in the _config file itself.

**Comment by jgclark** [comment]
Any update on this, @abcitygit ?

**Comment by abcitygit** [comment]
Hi,

Got it to work. I was missing the first character of the API key for the weather API. I went back and re-copied it and added curly brackets before and after each of the three code blocks under _configuration and it worked. I also hadn’t realized I needed to add a - before the longitude to get the correct weather. 

Thanks for following up and THANK YOU for making NotePlan! I’ve been a longtime Evernote user and then switched to DEVONthink and have been wanting a Markdown based/calendar integrated notebook. NotePlan 3 is a game changer with the integration of plugins across Mac and iOS.

Thank you!
Andrew
